pkg/cloudprovider/providers/openstack/openstack_instances.go

```diff
@@ -188,19 +188,17 @@ func (i *Instances) InstanceType(name types.NodeName) (string, error) {
 }
 
 func srvInstanceType(srv *servers.Server) (string, error) {
-	val, ok := srv.Flavor["name"]
-
-	if !ok {
-		return "", fmt.Errorf("flavor name not present in server info")
-	}
-
-	flavor, ok := val.(string)
-
-	if !ok {
-		return "", fmt.Errorf("flavor name is not a string")
+	keys := []string{"name", "id", "original_name"}
+	for _, key := range keys {
+		val, found := srv.Flavor[key]
```

Comment:
Minor style suggestion:  It's ever-so-slightly-more-idiomatic to inline trivial checks like these:

```go
if val, found := srv.Flavor[key]; found {
   if flavor, ok := val.(string); ok {
      return flavor, nil
   }
}
```